scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -18,7 +18,7 @@ import org.scalameta.{unreachable, debug}
 import scala.compat.Platform.EOL
 
 object TreeSyntax {
```

Comment:
Our current prettyprinter is outrageously inadequate - it is both slow and buggy - so I agree that it should be rewritten from scratch. However, I'm not sure about the exact strategy.

My concern about using `typelevel/paiges` is that the 3-5x slowdown advertised in the readme is not really competitive (even though the readme claims the opposite). It would be a shame to invest significant time into a rewrite only to discover that the new version is inadequate, too.

@olafurpg @DavidDudson What do you think?

/cc @densh, since he seems to have experimented with prettyprinters a lot with native. Denys, I see that you're currently using a dumb imperative prettyprinter. Are you happy with it? Do you plan to change it in the future?